lib/web/apps.go

```diff
@@ -357,15 +358,20 @@ func (h *Handler) proxyDNSNames() (dnsNames []string) {
 	return dnsNames
 }
 
-// removeTCPApps filters TCP apps out of the list of app servers.
-// TCP apps are filtered out because they are not accessible from the web UI.
-// It returns the HTTP apps and the number of TCP apps that were removed.
-func removeTCPApps(appServers []types.AppServer) (apps types.Apps, numExcluded int) {
+// removeUnsupportedApps filters unsupported (TCP, Cloud API-only) apps out of the list of app servers.
+func removeUnsupportedApps(appServers []types.AppServer) (apps types.Apps) {
 	for _, server := range appServers {
+		a := server.GetApp()
+		// Skip over API-only Cloud apps
+		if strings.HasPrefix(a.GetURI(), "cloud://") {
```

Comment:
what if the Cloud is `AWS` added by [a.URI = fmt.Sprintf("cloud://%v", a.Cloud)](https://github.com/gravitational/teleport/pull/18675/files#diff-054e31280a78f4aea4759cd740f8d8bcecf563679d7fe6b79991fdb944e8c611R1113) where CLI and Console access uses the same app ?

Reply:
If the Teleport admin wants to expose the AWS console as App, then they must set the URI to console for the chosen instance of AWS (public/gov cloud/China). Otherwise, the interpretation is that they don't want it to be exposed.

This is open to interpretation, and we may want to change how this detail works in the future. For existing configurations things work exactly as before.